src/StreamJsonRpc.Shared/JsonRpc.cs

```diff
@@ -655,6 +662,82 @@ protected virtual Task<ReturnType> InvokeCoreAsync<ReturnType>(int? id, string t
+        /// </summary>
+        /// <param name="target">Object to reflect over and analyze its methods.</param>
+        /// <returns>Dictionary which maps a request method name to its clr method name.</returns>
+        private static Dictionary<string, string> GetRequestMethodToClrMethodMap(object target)
+        {
+            // Create a dictionary to map clr method names to <see cref="JsonRpcMethodAttribute" /> values, used to detect that same method shouldn't have multiple attribute values.
+            var clrMethodToRequestMethodMap = new Dictionary<string, string>(StringComparer.Ordinal);
+
+            // Create a dictionary to map <see cref="JsonRpcMethodAttribute" /> values to clr method names, used to detect that the same attribute value doesn't map to multiple methods.
+            var requestMethodToClrMethodMap = new Dictionary<string, string>(StringComparer.Ordinal);
+
+            if (target != null)
+            {
+                for (TypeInfo t = target.GetType().GetTypeInfo(); t != null; t = t.BaseType?.GetTypeInfo())
+                {
+                    HashSet<string> clrMethodsWithoutReplacementName = new HashSet<string>(StringComparer.Ordinal);
+
+                    foreach (MethodInfo method in t.DeclaredMethods)
+                    {
+                        var attribute = (JsonRpcMethodAttribute)method.GetCustomAttribute(typeof(JsonRpcMethodAttribute));
+                        if (attribute == null)
+                        {
+                            clrMethodsWithoutReplacementName.Add(method.Name);
+                            if (clrMethodToRequestMethodMap.ContainsKey(method.Name))
+                            {
+                                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.OverloadsMissingMethodAttribute, method.Name, nameof(JsonRpcMethodAttribute)));
+                            }
+                        }
+                        else
+                        {
+                            if (clrMethodsWithoutReplacementName.Contains(method.Name))
+                            {
+                                throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.OverloadsMissingMethodAttribute, method.Name, nameof(JsonRpcMethodAttribute)));
+                            }
+
+                            string attributeValue;
+                            if (clrMethodToRequestMethodMap.TryGetValue(method.Name, out attributeValue))
+                            {
+                                if (!string.Equals(attribute.Name, attributeValue, StringComparison.Ordinal))
+                                {
+                                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Resources.ConflictingMethodNameAttribute, method.Name, nameof(JsonRpcMethodAttribute)));
```

Comment:
Tip: Your `throw new ArgumentException(string.Format(CultureInfo.CurrentCulture,` line could be reduced to start with just `Requires.Fail(`. It takes care of the throw, the string.Format and CultureInfo for you.